### DISCUSSION TITLE
SQL Server Container doesn't run entrypoint.sh on Azure

### DISCUSSION THREAD

**Discussion by Boggin** [microsoft/aspire#6486] [Q&A]
My AppHost/Program.cs (net8.0, Aspire 8) has the following configuration of a SQL Server Container:

```csharp
var sql = builder.AddSqlServer("sql")
      .WithBindMount("./sqlserverconfig", "/usr/config")
      .WithBindMount("../MyApp.Web/data/sqlserver", "/docker-entrypoint-initdb.d")
      .WithEntrypoint("/usr/config/entrypoint.sh")
      .PublishAsContainer()
      .AddDatabase("mydb", "MYDB");
```

This works during local development. There's an `init.sql` script in data/sqlserver that creates the db.

When deployed the `WithBindMount()` points to a folder that isn't on the host.

What's the best way to create the database from the `init.sql` script when publishing?

Should I get my files into those file shares during deployment? Is `.WithDockerfile()` a good way to do this (where I create an image with those files)?

**Comment by davide-podda** [comment]
yes, i solve this using ef migration. see database migration in the aspire-sample

**Reply by Boggin** [reply]
I suppose this wasn't clear from my question but I'm not using EF. I'm using Dapper, thus the init.sql to set up the db. You are right, however, that the SQL db could be set up by the application code on start-up, like an EF migration. Probably with a separate worker service to keep it the responsibility clear as I have more that one microservice using the db. I'll give that a try.

**Comment by davidfowl** [comment]
It not running on azure might be an issue with the volume. What errors do you see (they may not show up in the aspire dashboard)
